### DISCUSSION TITLE
x-data in loop empty ( Laravel )

### DISCUSSION THREAD

**Discussion by stallyons-developer1** [alpinejs/alpine#2959] [7. Bugs (archived)]
Here is my code, it seems perfect but somehow it's not working and even in the chrome dev tool when I inspect each component the x-data is empty.
```
@foreach($invoices as $invoice)

<div class="dropdown" x-data="{open : false}">

	<button @click="open = !open" class="btn btn-secondary dropdown-toggle" type="button">
		<i class="fas fa-ellipsis-v"></i>                                            
	</button>

 	 <div class="dropdown-menu" x-show="open" x-transition>
		...content
	</div>

</div>

@endforeach
```

but outside of the loop, it works fine

```
<div class="dropdown" x-data="{open : false}">

	<button @click="open = !open" class="btn btn-secondary dropdown-toggle" type="button">
		<i class="fas fa-ellipsis-v"></i>                                            
	</button>

 	 <div class="dropdown-menu" x-show="open" x-transition>
		...content
	</div>

</div>

<div class="dropdown" x-data="{open : false}">

	<button @click="open = !open" class="btn btn-secondary dropdown-toggle" type="button">
		<i class="fas fa-ellipsis-v"></i>                                            
	</button>

 	 <div class="dropdown-menu" x-show="open" x-transition>
		...content
	</div>

</div>

```

**Comment by SimoTod** [comment]
The loop is generated server side (I assume you are using blade) so Alpine doesn't really know about that. I believe the `@` needs escaping or you can use the long version `x-on:click`

**Reply by stallyons-developer1** [reply]
Yes, I am using blade, and I have used x-on:click that's also not working. @SimoTod

**Reply by SimoTod** [reply]
just copied your code on Laravel playground. Alpine works correctly. It looks like you are using bootstrap, from a quick check `dropdown-menu` has `display: none` so it does not show because of bootstrap. I think you. need to add the class `show` to the same div: `<div class="dropdown-menu show" x-show="open" x-transition>`
